portfolio/src/main/webapp/index.html

```diff
@@ -141,13 +141,24 @@ <h1>Andrzej Podobinski</h1>
             </ul>
             <button onclick="nextComments()">Load older</button>
         </div>
-        <form action="/data" method="POST">
-                <label for="user-name">Name: </label>
-                <input type="text" id="user-name" name="user-name"><br>
+        <div id="login-div">
+            <h4 id="login-message"></h4>
+            <button id="login-button" onclick="">LOG IN</button>
+            <button id="logout-button" onclick="">LOG OUT</button>
+        </div>
+        <form id="comment-form" action="/data" method="POST">
+                <input type="hidden" id="user-name" name="user-name" readonly><br>
                 <label for="comment">Comment: </label>
                 <textarea type="text" id="comment" name="comment"></textarea><br>
                 <input onclick="newestComments()" type="submit"/>
         </form>
+        <form id="nickname-form" action="/nickname" method="POST">
+            <label for="nickname">Nickname: </label>
+            <input type="text" id="nickname" name="nickname"><br>
+            <input type="submit"/>
+        </form>
```

Comment:
Although the robust way is we should have server-side validation as well, but shall we add some basic client-side validation, such as requiring the value to exist and not be whitespace? See: https://developer.mozilla.org/en-US/docs/Learn/Forms/Form_validation

Reply:
Good suggestion, I've added some basic html5 authentication.